File: common/src/main/java/org/valkyrienskies/clockwork/content/contraptions/phys/bearing/PhysBearingUpdateData.java
package org.valkyrienskies.clockwork.content.contraptions.phys.bearing;

import org.joml.Vector3d;
import org.valkyrienskies.core.apigame.constraints.VSConstraintAndId;
import org.valkyrienskies.core.apigame.constraints.VSFixedOrientationConstraint;
import org.valkyrienskies.core.apigame.constraints.VSHingeOrientationConstraint;

public record PhysBearingUpdateData(double bearingAngle, float bearingRPM, boolean locked, VSHingeOrientationConstraint hingeConstraint, VSFixedOrientationConstraint angleConstraint) {
}

File: common/src/main/kotlin/org/valkyrienskies/clockwork/content/contraptions/phys/bearing/data/PhysBearingUpdateData.kt
package org.valkyrienskies.clockwork.content.contraptions.phys.bearing.data

import org.valkyrienskies.core.apigame.constraints.VSFixedOrientationConstraint
import org.valkyrienskies.core.apigame.constraints.VSHingeOrientationConstraint

data class PhysBearingUpdateData(
        val bearingAngle: Double,
        val bearingRPM: Float,
        val locked: Boolean,
        val hingeConstraint: VSHingeOrientationConstraint?,
        val angleConstraint: VSFixedOrientationConstraint?
)